Automattic/sensei
# Add usage tracking to features step

Fixes #3051

### Changes proposed in this Pull Request

* Add the `sensei_setup_wizard_features_continue` event
 * Triggered when the features step is done (after installations if selected).
* Add the `setup_wizard_features_learn_more` event to link clicks on the Features page
* Save and restore the selected features - show submit progress before installation confirm dialog

This uses the existing jQuery-based event logging code on the frontend for now. (`window.sensei_log_event` in `event-logging.js`)

We also update the `window.sensei_event_logging.enabled` setting when the user changes their usage tracking preference in the Welcome step, so that it immediately takes effect, and the log events are only sent when it's on. 

Also adds a small helper for link tracking that makes sure that links opened by middle clicking are also registered.

### Testing instructions

* Enable usage tracking
* Open Features step
* Click on `Learn more` links - related events with the correct slug should show up
* Select some features
* Click on `Continue`, then `I'll do it later`, or `Install now`, then `Continue` when it's done - event with the selected slugs should show up
* Selecting no features - event should show up after clicking the `Continue` button

--
* Change usage tracking preference on the welcome step
* Step to features, click on Learn more links
* Event log request should only be sent when tracking is enabled.
